daffodil-japi/src/test/java/org/apache/daffodil/example/TestJavaAPI.java

```diff
@@ -48,9 +49,31 @@
 import org.apache.daffodil.japi.logger.ConsoleLogWriter;
 import org.apache.daffodil.japi.logger.LogLevel;
 import org.apache.daffodil.japi.io.InputSourceDataInputStream;
+import org.xml.sax.SAXNotRecognizedException;
+import org.xml.sax.SAXNotSupportedException;
+import org.xml.sax.XMLReader;
+
+import javax.xml.XMLConstants;
 
 public class TestJavaAPI {
 
+    /**
+     * Best practices for XML loading are to turn off anything that could lead to
+     * insecurity.
+     *
+     * This is probably unnecessary in the case of these tests, but as these tests
+     * are also used to illustrate API usage, this exemplifies best practice.
+     */
+    public static void setSecureDefaults(XMLReader xmlReader)
+            throws SAXNotSupportedException, SAXNotRecognizedException {
+        xmlReader.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        // since we're not really sure what they mean by secure processing
+        // we make doubly sure by setting these ourselves also.
+        xmlReader.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+        xmlReader.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+        xmlReader.setFeature("http://xml.org/sax/features/external-general-entities", false);
+    }
+
```

Comment:
I'm not sure if this is officially, but, I only think of things in japi/sapi as being part of the public API. Other public classes/functions people can use, but they aren't neesssarily stable and we won't make an attempt at backwards compatibility.

Since XMLUtils isn't part of sapi/japi, I would prefer that we not use it in japi/sapi tests since we often point people towards these tests for how to use API. That way we aren't suggesting the use of non public API.

Reply:
I have a slightly different suggestion.  If we ever make secure XML loading part of the published API, let's publish DaffodilSAXParserFactory instead of XMLUtils.  I know there's also JDOM parsing and so on, but we don't have to offer every way to load XML securely in the published API, only the way which we recommend as the best way.  When people use DaffodilSAXParserFactory, they insure that the SAXParser they get is exactly the Xerces one and its XMLReader is secure just by changing their first line of code.  That's the easiest way to load XML securely - just a one-line change :).